### ISSUE TITLE
CSS highlight inside PHP files

### ISSUE THREAD

**Issue by diego-vieira** [microsoft/vscode#33519]
- VSCode Version: 1.15.1
- OS Version: macOS 10.12.6

Steps to Reproduce:

1. Open a new PHP file with this content:
```php
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <meta http-equiv="X-UA-Compatible" content="ie=edge">
    <title>Document</title>

    <style>
    .test {
        color: #fff;
    }

    <?php
    echo $css;
    ?>
    </style>
</head>
<body>
    
</body>
</html>
```
2. Notice that after the `$css;` the highlight of CSS expands to the end of the file.

Reproduces without extensions: Yes

**Comment by roblourens** [comment]
PHP in <style> tags is colored correctly in Atom but not vscode using the same grammar...


```
<html>
<script>
    <?php
        $foo = "bar";
    ?>
</script>
<style>
    <?php
        $foo = "bar";
    ?>
</style>

<!-- foo -->
</html>
```

![image](https://user-images.githubusercontent.com/323878/29883226-743fdee2-8d64-11e7-8c09-d4ce06f741a4.png)

**Comment by jens1o** [comment]
what can be done to bring this to the deck?

**Comment by swashata** [comment]
Hi,

I just switched from SublimeText to VSCode and this is becoming a blocker for me. The following PHP code is the least which can reproduce the error in `.php` file with `PHP` Language.

```php
<?php

function some_test_with_css( $my_css ) {
	?>
<style type="text/css">
	<?php echo $my_css; ?>
</style>
	<?php
}

function some_other_php_function() {
	// Do something here
	$my_variable = 'foo';
	$bar = 'bar';
	echo join( ' ', [ $my_variable, $bar ] );
}
```

<img width="1364" alt="screen shot 2018-03-29 at 10 55 48 am" src="https://user-images.githubusercontent.com/1623381/38071641-1d15881a-3340-11e8-951e-d5927c2d920a.png">


Please see the screenshot. Anything after `<style type="text/css"` gets `meta.selector.css` and `source.css` scope. Hence the issue. Is it possible to resolve it please? Or is there any fix in the nightly version which I can try? Any pointer on using the atom package directly to see if it fixes things?

/cc @jens1o @roblourens

**Comment by swashata** [comment]
I guess this is a duplicate of #20488

**Comment by skoshy** [comment]
This is an issue for me as well, even on the latest VSCode version.

Screenshot, no extensions enabled:
![2018-05-21 12 35 38](https://user-images.githubusercontent.com/369825/40318875-26521e10-5cf4-11e8-8450-8c37e27a0403.png)

**Comment by AmirHosseinKarimi** [comment]
Hi, I have this problem too in Debian 9.
Version: 1.28.0-insider
Commit: b4d09d27509f2c023f63d4315aa966a5ec7581e9
Date: 2018-09-27T14:42:25.882Z
Electron: 2.0.9
Chrome: 61.0.3163.100
Node.js: 8.9.3
V8: 6.1.534.41
Architecture: x64

![screenshot from 2018-10-10 19-12-53](https://user-images.githubusercontent.com/17345129/46749100-ee166680-ccc1-11e8-985e-9df4b4fcd2d8.png)

**Comment by AmirHosseinKarimi** [comment]
@swashata Thanks, but look at line 282, I used php code inside a property name but it does not highlighted. This is my problem...

**Comment by swashata** [comment]
ah, I don't know how to fix that :(

**Comment by wayheming** [comment]
+1 up

**Comment by wayheming** [comment]
@swashata thanks!

**Comment by AnsonYeung** [comment]
#65649
Why css is different from js?

**Comment by Sumanai** [comment]
VSCode 1.33.1, The problem is still playing. Any error in the CSS part breaks the backlight of the entire file down. Example
```
<style>
.

</style>

<?php

$var = '';
```

**Comment by JuegOStrower** [comment]
Version 1.35.1, the bug hasn't been fixed yet!

**Comment by SerdarDemirbas** [comment]
This is big problem in same file. Ver. 1.34.0 continues the problem.

**Comment by connorrooney** [comment]
Ver 1.36.1 is still producing the bug.

**Comment by roblourens** [comment]
Will be fixed by #80657
